server/src/main/java/io/crate/statistics/ReservoirSampler.java

```diff
@@ -89,20 +93,59 @@ public final class ReservoirSampler {
     private CircuitBreakerService circuitBreakerService;
     private final IndicesService indicesService;
 
+    private volatile ByteSizeValue readLimit;
+    private volatile ByteSizeValue updatedReadLimit;
+    private volatile boolean analyzing;
+
+    private final RateLimiter rateLimiter;
+    private long bytesSinceLastRateLimit = 0;
+
     @Inject
     public ReservoirSampler(ClusterService clusterService,
                             NodeContext nodeCtx,
                             Schemas schemas,
                             CircuitBreakerService circuitBreakerService,
-                            IndicesService indicesService) {
+                            IndicesService indicesService,
+                            Settings settings) {
         this.clusterService = clusterService;
         this.nodeCtx = nodeCtx;
         this.schemas = schemas;
         this.circuitBreakerService = circuitBreakerService;
         this.indicesService = indicesService;
+
+        this.readLimit = STATS_SERVICE_THROTTLING_SETTING.get(settings);
+        this.rateLimiter = new RateLimiter.SimpleRateLimiter(readLimit.getMbFrac());
+        this.updatedReadLimit = readLimit;
+        this.analyzing = false;
+
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(
+            STATS_SERVICE_THROTTLING_SETTING, this::updateReadLimit);
+    }
+
+    private void updateReadLimit(ByteSizeValue newReadLimit) {
+        boolean isAnalyzing = analyzing; // One volatile read.
+        if (isAnalyzing == false) {
+            readLimit = newReadLimit;
+            rateLimiter.setMBPerSec(readLimit.getMbFrac());
+        } else {
+            updatedReadLimit = newReadLimit; // One volatile write. Let running ANALYZE operation finish with the previous value.
+        }
```

Comment:
I suggest to set the rateLimiter to NULL to disable it to avoid useless calls.

Reply:
covered disabled case in https://github.com/crate/crate/pull/13046/commits/a15139bc2494b2c000c8c4ecf20c3eb624988367
I didn't set limiter to null since it might be updated in `updateReadLimit` and can cause NPE. 
Currently maubepause is not called if setting is disabled - later on when we extend grammar, getSamles will get additional flag and it will be also aded to the check.